### PR TITLE
 fix IntrusiveQueue::remove: several bugs that led to improper state of the queue

### PR DESCRIPTION
**Pull request by bkueng**
@limhyon noticed that tfmini was not working anymore on master, and traced it down to https://github.com/PX4/Firmware/blob/004d916095321c01c08719e12ca971d1e45a5453/src/drivers/distance_sensor/tfmini/TFMINI.cpp#L248 
The driver was suddenly not scheduled anymore, and removing the line 'fixed' it.

The underlying problem is however an incorrect implementation of `IntrusiveQueue::remove`, and it got revealed with https://github.com/PX4/Firmware/pull/14331.

Also, tfmini was scheduled at 10kHz which seems excessive given the sensor outputs data at 100Hz. It is now reduced to 500Hz.

@limhyon can you test this?

### CODE REVIEW

**Comment by dagar** [comment]
Nice find. I was actually still hoping to get rid of our own containers at some point.

File: src/drivers/distance_sensor/tfmini/TFMINI.cpp [hunk 1]
```diff
@@ -224,7 +224,7 @@ void
 TFMINI::start()
 {
 	// schedule a cycle to start things
-	ScheduleOnInterval(100_us);
+	ScheduleOnInterval(2_ms);
```
Comment (dagar) [inline src/drivers/distance_sensor/tfmini/TFMINI.cpp]: Even this seems excessive. Wondering if it was supposed to be 100_ms in the first place?

Comment (bkueng) [reply src/drivers/distance_sensor/tfmini/TFMINI.cpp]: Agreed. Reduced to 7ms.

**Comment by limhyon** [comment]
Will do test soon.

**Pull request merged**